packages/proto-loader/src/index.ts

```diff
@@ -368,6 +379,54 @@ export function loadSync(
   return createPackageDefinition(root, options!);
 }
 
+export function loadFileDescriptorSet(
+  descriptorSet:
+    | Buffer
+    | ReturnType<typeof descriptor.FileDescriptorSet.toObject>,
+  options?: Options
+): PackageDefinition {
+  type DecodedDescriptorSet = Protobuf.Message<descriptor.IFileDescriptorSet> &
+    descriptor.IFileDescriptorSet;
+
+  options = options || {};
+
+  let decodedDescriptorSet: DecodedDescriptorSet;
+  if (Buffer.isBuffer(descriptorSet)) {
+    decodedDescriptorSet = descriptor.FileDescriptorSet.decode(
+      descriptorSet
+    ) as DecodedDescriptorSet;
+  } else {
+    decodedDescriptorSet = descriptor.FileDescriptorSet.fromObject(
+      descriptorSet
+    ) as DecodedDescriptorSet;
+  }
+
+  const root = (Protobuf.Root as Protobuf.RootConstructor).fromDescriptor(
+    decodedDescriptorSet
+  );
+  root.resolveAll();
+  return createPackageDefinition(root, options);
+}
+
+export function loadFileDescriptorSetFile(
```

Comment:
I don't think this function is necessary. Its functionality is just "Read a file and pass the contents through a different exported function". That's not specific to this library, so I think we should just let the user determine where the data comes from.

That contrasts with the `load` and `loadSync` functions, in which dependency resolution across files is an essential part of their functionality.